nifi-nar-bundles/nifi-slack-bundle/nifi-slack-services-api/src/main/java/org/apache/nifi/processors/slack/controllers/SlackConnectionService.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.slack.controllers;
+
+import java.util.function.Consumer;
+
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.controller.ControllerService;
+import org.apache.nifi.processor.Processor;
+
+@CapabilityDescription("This service provides a connection to slack")
+public interface SlackConnectionService extends ControllerService {
```

Comment:
One of benefits of having ControllerService is sharing common configurations or providing connection pooling. It seems ListenSlack is the only processor that uses this CS. What is the benefit of having CS in this case? Other Slack processors have its own API token properties.. Is there any plan to share this CS with other processors?

Reply:
Thanks for the explanation. That makes sense :)